implementations/rust/ockam/ockam_api/src/nodes/service/secure_channel.rs

```diff
@@ -127,41 +157,37 @@ impl NodeManager {
             body.channel
         );
 
-        let identity = self.identity()?.async_try_clone().await?;
+        let identity = self.identity()?;
 
         let sc_address = Address::from(body.channel.as_ref());
 
         debug!(%sc_address, "Deleting secure channel");
 
-        // Best effort to tear down the channel.
-        let _ = identity.stop_secure_channel(&sc_address).await;
-
-        // Remove both the Address and Route entries from the registry.
-        let sc_addr = sc_address.clone().into();
-        let key_addr = IdentityRouteKey::new(&identity, &sc_addr).await?;
-        let res = if let Some(v) = self.registry.secure_channels.remove(&key_addr) {
-            trace!(%sc_addr, "Removed secure channel");
-            let sc_route = v.route();
-            let key_route = IdentityRouteKey::new(&identity, sc_route).await?;
-            self.registry.secure_channels.remove(&key_route);
-            trace!(%sc_route, "Removed secure channel");
-            Some(sc_address)
-        } else {
-            trace!(%sc_addr, "No secure channels found for the passed address");
-            None
+        let res = match identity.stop_secure_channel(&sc_address).await {
+            Ok(()) => {
+                trace!(%sc_address, "Removed secure channel");
+                self.registry.secure_channels.remove_by_addr(&sc_address);
+                Some(sc_address)
+            }
+            Err(err) => {
+                trace!(%sc_address, "Error removing secure channel: {err}");
+                None
+            }
```

Comment:
To improve the logged info, maybe we could check if the `sc_address` is in the registry before trying to stop it, and log a "Not found" kind of message instead of the current "Error" kind of message.

```rust
let res = if !self.registry.secure_channels.contains(&sc_address) {
    trace!(%sc_address, "No secure channel found for the passed address");
    None
} else {
    match identity.stop_secure_channel(&sc_address).await {
        ...
    }
};
```

Reply:
Well, channels in registry and actual running channels should hopefully match 1:1, in practice that's not guaranteed. I think if we successfully stopped the channel, and registry didn't contain such channel, we're in perfectly valid state. Channel is stopped and not present in the registry. Yay